File: c/1512_num_of_pairs.c
int numIdenticalPairs(int* nums, int numsSize){
    //its a straight solution (brute force)
    //NB for myself: шт the future, think about how this can be implemented with hashtable
    int result = 0;
    for (int i = 0; i < numsSize; i++)
      for (int j = i + 1; j < numsSize; j++)
        result += nums[i] == nums[j];
    return result;
}

File: python/1512_num_of_pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        # using defaultdict for counting
        temp_hash = defaultdict(int)
        result = 0
        for i in nums:
            result += temp_hash[i]
            temp_hash[i] += 1
        return result